pkg/detectors/stripepaymentintent/stripepaymentintent.go

```diff
@@ -0,0 +1,203 @@
+package stripepaymentintent
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	Client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	clientSecretPat   = regexp.MustCompile(`\b(pi_[a-zA-Z0-9]{24}_secret_[a-zA-Z0-9]{25})\b`)
+	secretKeyPat      = regexp.MustCompile(`\b([rs]k_live_[a-zA-Z0-9]{20,247})\b`)
+	publishableKeyPat = regexp.MustCompile(`\b(pk_live_[a-zA-Z0-9]{20,247})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"_secret_"}
+}
+
+func (s Scanner) getClient() *http.Client {
+	if s.Client != nil {
+		return s.Client
+	}
+	return defaultClient
+}
+
+// FromData will find and optionally verify Stripe Payment Intent secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) ([]detectors.Result, error) {
+	dataStr := string(data)
+
+	clientSecrets := extractMatches(clientSecretPat, dataStr)
+	secretKeys := extractMatches(secretKeyPat, dataStr)
+	publishableKeys := extractMatches(publishableKeyPat, dataStr)
+
+	// Only proceed if we have client secrets AND at least one key
+	totalKeys := len(secretKeys) + len(publishableKeys)
+	if len(clientSecrets) == 0 || totalKeys == 0 {
+		return nil, nil
+	}
+
+	results := make([]detectors.Result, 0, len(clientSecrets)*totalKeys)
+	client := s.getClient()
```

Comment:
We can move this to the verify block.

Reply:
Then we will have to call the function two times, as we are doing two types of verification.

Reply:
updated.